### PR TITLE
fix(platforms-serverless): prevent lambda and sst flakiness

### PR DESCRIPTION
**Pull request by millsp**
Fixes flakiness for `lambda` tests and `serverless-framework-lambda` by separating `library` and `binary` deployments into two separate functions.

### CODE REVIEW

File: platforms-serverless/serverless-framework-lambda/serverless.yml [hunk 1]
```diff
@@ -1,8 +1,8 @@
-service: e2e-tests-serverless-lambda
+service: platforms-serverless-slsf-${env:PRISMA_CLIENT_ENGINE_TYPE}
 
 provider:
   name: aws
-  runtime: nodejs16.x
+  runtime: nodejs18.x
```
Comment (Jolg42) [inline platforms-serverless/serverless-framework-lambda/serverless.yml]: I also want to switch to v18 but we still need to test v16?

File: platforms-serverless/lambda/run.sh [hunk 2]
```diff
@@ -9,9 +9,12 @@ x="$AWS_SECRET_ACCESS_KEY"
...
 pnpm prisma generate
-
 pnpm tsc
 
-sh update-code.sh
+rm -rf lambda.zip
+zip --symlinks -r lambda.zip index.js prisma/schema.prisma node_modules/@prisma/client node_modules/.pnpm/@prisma+client*
+du -b ./lambda.zip
+
+aws lambda update-function-configuration --function-name "platforms-serverless-lambda-$PRISMA_CLIENT_ENGINE_TYPE" --runtime nodejs18.x --environment "Variables={DATABASE_URL=$DATABASE_URL}" --timeout 10
```
Comment (Jolg42) [inline platforms-serverless/lambda/run.sh]:
```suggestion
aws lambda update-function-configuration --function-name "platforms-serverless-lambda-$PRISMA_CLIENT_ENGINE_TYPE" --runtime nodejs16.x --environment "Variables={DATABASE_URL=$DATABASE_URL}" --timeout 10
```

File: platforms-serverless/serverless-framework-lambda/serverless.yml (hunk 1 above)
Comment (Jolg42) [inline platforms-serverless/serverless-framework-lambda/serverless.yml]:
```suggestion
  runtime: nodejs16.x
```

**Review by Jolg42 — APPROVED** [review]
Thank you!

**Comment by janpio** [comment]
What is the actual change here that matters?
I could only identify a few renames.

**Comment by Jolg42** [comment]
@janpio Instead of one function we have 2 now, see https://github.com/prisma/ecosystem-tests/pull/3909/files#diff-02553ce737b2aea2c7d1292bdef061a1185d64331bf94a40e2c104f7fa8e496fR19-R20

**Pull request merged**